BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    /// <param name="bits">A 16 bit character array (<see cref="System.String"/>).</param>
+    public BitList(string bits)
+    {
+        var bytesList = new List<byte[]>();
+        foreach (var num in bits)
+            bytesList.Add(BitConverter.GetBytes(num));
+        var byteList = new List<byte>();
+        foreach (var bytes in bytesList)
+        {
+            foreach (var num in bytes)
+                byteList.Add(num);
+        }
+        this.bits = new BitArray(byteList.ToArray());
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit character array of array.</param>
+    public BitList(string[] bits)
+    {
+        var _bytesList = new List<byte[]>();
+        var _byteList = new List<byte>();
+        foreach (var str in bits)
+        {
+            var bytesList = new List<byte[]>();
+            foreach (var chr in str)
+                bytesList.Add(BitConverter.GetBytes(chr));
+            var byteList = new List<byte>();
+            foreach (var bytes in bytesList)
+            {
+                foreach (var num in bytes)
+                    byteList.Add(num);
+            }
+            _bytesList.Add(byteList.ToArray());
+        }
+        foreach (var bytes in _bytesList)
+        {
+            foreach (var num in bytes)
+                _byteList.Add(num);
+        }
+        this.bits = new BitArray(_byteList.ToArray());
+    }
+
+    IEnumerator<bool> IEnumerable<bool>.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    /// <summary>
+    /// Get the index of a bit in this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    /// <returns>The <paramref name="bit"/> index.</returns>
+    public int IndexOf(bool bit)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (bit == this[i])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Get the index of a list of bit in this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public int IndexOf(BitList bits)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (bits == this[i, bits.Count])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Clone this instance in a new object.
+    /// </summary>
+    public object Clone()
+    {
+        return this;
+    }
+
+    /// <summary>
+    /// Return true if the specified BitList is equal to this istance.
+    /// </summary>
+    /// <param name="other">A list of bit.</param>
+    public bool Equals(BitList other)
+    {
+        return Equals(this, other);
+    }
+
+    /// <summary>
+    /// Return true if this instance contains the specified bit.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public bool Contains(bool bit)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'bit' to 'item' to match the interface declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)